lib/InteractiveViewer.ml

```diff
@@ -0,0 +1,126 @@
+open Nottui
+module W = Nottui_widgets
+open Lwd_infix
+
+type view_mode = SideBySide | Normal
+
+let view_mode : view_mode Lwd.var = Lwd.var Normal
+
+let toggle_view_mode () : unit =
+  match Lwd.peek view_mode with
+  | Normal -> Lwd.set view_mode SideBySide
+  | SideBySide -> Lwd.set view_mode Normal
+
+let help_visible = Lwd.var false
+let quit = Lwd.var false
+
+let toggle_help_visibility () =
+  Lwd.set help_visible (not (Lwd.peek help_visible))
+
+let view (patches : Patch.t list) =
+  let z_patches_var : Patch.t Zipper.t Lwd.var =
+    match Zipper.zipper_of_list patches with
+    | Some z -> Lwd.var z
+    | None -> failwith "zipper_of_list: empty list"
+  in
+  let hunks_ui =
+    let$ mode = Lwd.get view_mode and$ z_patches = Lwd.get z_patches_var in
+    match mode with
+    | Normal -> HunkView.current_hunks z_patches
+    | SideBySide -> HunkView.current_hunks_side_by_side z_patches
+  in
+  let curr_scroll_state = Lwd.var W.default_scroll_state in
+  let change_scroll_state _action state =
+    let off_screen = state.W.position > state.W.bound in
+    if off_screen then
+      Lwd.set curr_scroll_state { state with position = state.W.bound }
+    else Lwd.set curr_scroll_state state
+  in
+  let project_op (op : Patch.t Zipper.t -> Nottui.ui) : ui Lwd.t =
+    Lwd.map (Lwd.get z_patches_var) ~f:op
```

Comment:
@Julow I rewrote it so more explicit and easier to follow

Reply:
I don't think this is easier to follow than inlining the `Lwd.map` directly where it's needed. I'd say introducing the `project_op` makes things less explicit.
We can keep it like that for now though.

Reply:
I think it is because to:
- Call `Lwd.get` once to get an `Lwd.t` value that reacts on `z_patches_var` changes
- Inline this function on places where it is used.

Reply:
Added